### PR TITLE
Piece-wise constant control functions and total-variation penalty term

### PR DESCRIPTION
**Pull request by andersp**
This PR includes a new class for piece-wise constant control functions called BSpline0. It also includes a new penalty term that suppresses sudden variations in the control amplitude, when the BSpline0 basis is used during optimization. Functionality has also been added to use the new functionality from the Python interface, see examples/pythoninterface/example_spline0.py.

### CODE REVIEW

**Review by steffi7574 — CHANGES_REQUESTED** [review]
Looks good! I added few comments.

File: include/controlbasis.hpp [hunk 1]
```diff
@@ -125,6 +130,37 @@ class Step : public ControlBasis {
...
+        // double *tcenter;                  // vector of basis function center positions
+        double width;                     // support of each basis function (m*dtknot)
+
+        /* Evaluate the bspline basis functions B_l(tau_l(t)) NOT USED */
+        // double bspl0(int id, double t);
+
+    public:
+        BSpline0(int nsplines, double tstart, double tstop, bool enforceZeroBoundary);
+        ~BSpline0();
+
```
[the remark this answers is not in the document]
Comment (andersp) [reply include/controlbasis.hpp:151]: By definition, this function is always zero for t<0 and t>Tmax. Thus, there doesn't seem to be any need to enforce zero boundary conditions.

File: examples/pythoninterface/BS0-pert_params.dat
[the remark this answers is not in the document]
Comment (andersp) [reply examples/pythoninterface/BS0-pert_params.dat:1]: I have added another example to show how the control vector can be read from file with the new pcof0_filename option. This example needs the file BS0-pert_params.dat.

File: examples/pythoninterface/example_spline0.py
[the remark this answers is not in the document]
Comment (andersp) [reply examples/pythoninterface/example_spline0.py:1]: Agreed. Done.

**Comment by andersp** [comment]
It would make sense to move the description of the Tikhonov term to section 3.5, where all the other penalty terms are discussed.

**Pull request merged**